src/api/process-manager.ts

```diff
@@ -126,46 +133,51 @@ export function startBuild(data: any): Promise<any> {
       return insertBuildRun(data);
     })
     .then(() => getBuild(data.build_id))
-    .then(buildData => sendPendingStatus(buildData, buildData.id))
-    .then(() => {
-      return config.reduce((prev, cfg, i) => {
-        return prev.then(() => {
-          let dataJob = null;
-
-          return dbJob.insertJob({ data: JSON.stringify(cfg), builds_id: data.build_id })
-            .then(job => dataJob = job)
-            .then(() => getLastRunId(data.build_id))
-            .then(lastRunId => {
-              const jobRun = {
-                start_time: new Date,
-                status: 'queued',
-                build_run_id: lastRunId,
-                job_id: dataJob.id
-              };
-
-              return dbJobRuns.insertJobRun(jobRun);
-            })
-            .then(() => {
-              jobEvents.next({
-                type: 'process',
-                build_id: data.build_id,
-                job_id: dataJob.id,
-                data: 'jobAdded'
+    .then(buildData => {
+      sendPendingStatus(buildData, buildData.id)
+      .then(() => {
+        return config.reduce((prev, cfg, i) => {
+          return prev.then(() => {
+            let dataJob = null;
+
+            return dbJob.insertJob({ data: JSON.stringify(cfg), builds_id: data.build_id })
+              .then(job => dataJob = job)
+              .then(() => getLastRunId(data.build_id))
+              .then(lastRunId => {
+                const jobRun = {
+                  start_time: new Date,
+                  status: 'queued',
+                  build_run_id: lastRunId,
+                  job_id: dataJob.id
+                };
+
+                return dbJobRuns.insertJobRun(jobRun);
+              })
+              .then(() => {
+                jobEvents.next({
+                  type: 'process',
+                  build_id: data.build_id,
+                  job_id: dataJob.id,
+                  data: 'jobAdded'
+                });
+
+                return queueJob(data.build_id, dataJob.id);
               });
-
-              return queueJob(data.build_id, dataJob.id);
-            });
+          });
+        }, Promise.resolve());
+      })
+      .then(() => {
+        jobEvents.next({
+          type: 'process',
+          build_id: data.build_id,
+          repository_id: data.repositories_id,
+          data: 'buildAdded'
         });
-      }, Promise.resolve());
+      })
+      .then(() => getDepracatedBuilds(buildData))
+      .then(builds => Promise.all(builds.map(build => stopBuild(build))));
```

Comment:
question: will this force stop all related builds even if not running and write `failed` into database for all that builds?

Reply:
I'm preparing bug fix, where all builds gets updated end time, when they are in failed or success state. In that case, those were not returned here, only the ones with end_time null are returned here...

Reply:
okay, will this be included in that PR?

Reply:
Yes